src/EditorFeatures/Core/Implementation/IntelliSense/Helpers.cs

```diff
@@ -72,7 +140,60 @@ internal static IEnumerable<object> BuildClassifiedTextElements(ImmutableArray<T
             return paragraphs;
         }
 
-        internal static object CreateParagraphFromLines(IReadOnlyList<ClassifiedTextElement> lines)
+        private static void NavigateToQuickInfoTarget(string navigationTarget, Document document, IEnumerable<Lazy<IStreamingFindUsagesPresenter>> streamingPresenters)
+        {
+            var navigateToLinkService = document.Project.Solution.Workspace.Services.GetRequiredService<INavigateToLinkService>();
+            if (Uri.TryCreate(navigationTarget, UriKind.Absolute, out var absoluteUri))
+            {
+                navigateToLinkService.TryNavigateToLinkAsync(absoluteUri, CancellationToken.None);
+                return;
+            }
+
+            SymbolKeyResolution resolvedSymbolKey;
+            try
+            {
+                resolvedSymbolKey = SymbolKey.Resolve(navigationTarget, document.Project.GetCompilationAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None), cancellationToken: CancellationToken.None);
```

Comment:
oh ick... is this catch happening because we just have a string, and we're not even sure if it's a SymbolKey?  I *really* don't like that.  It's just sending corrupt data into some random API and basically needing it to throw to indicate "yeah... that's not from me"  Can we not have hte nav-targets be more strongly typed, so that we know if it's a link-uri, versus symbol-key?  If we do this, then we can just call this, without this heinous try/catch here.

Reply:
I agree; will review this in combination with https://github.com/dotnet/roslyn/pull/35667#discussion_r287448443.

Reply:
➡️ Filed follow-up issue #37500